code/modules/lootpanel/search_object.dm

```diff
@@ -0,0 +1,70 @@
+/**
+ * ## Search Object
+ * An object for content lists. Compacted item data.
+ */
+/datum/search_object
+	/// Client attached to the search_object
+	var/datum/weakref/client_ref
+	/// Weakref to the original object
+	var/datum/weakref/item_ref
```

Comment:
similar but with the qdeleting signal

Reply:
not sure what you meant here

Reply:
we listen for item QDELETING on the lootpanel datum, which owns this datum, and delete this datum when the item is deleted, so why does this need to be a weakref